lib/network_ssl/network_ssl_compat.c

```diff
@@ -63,6 +64,54 @@ network_ssl_compat_CTL_set_min_proto_version(SSL_CTX * ctx, int version)
 }
 #endif
 
+#ifdef NETWORK_SSL_COMPAT_CHECK_HOSTNAME
+/**
+ * network_ssl_compat_set1_host(ssl, hostname):
+ * Set expected name for hostname verification.
+ *
+ * COMPATIBILITY: provides SSL_set1_host().
+ */
+int
+network_ssl_compat_set1_host(SSL * ssl, const char * hostname)
+{
+
+	X509_VERIFY_PARAM * param;
+
+	param = SSL_get0_param(ssl);
+
+	/*
+	 * The docs don't say that this can happen, but better safe than sorry!
```

Comment:
Unnecessary snark.  And for that matter I'd say it's an unnecessary assertion; OpenSSL will core dump very fast if you pass NULL to X509_VERIFY_PARAM_set1_host.

Reply:
I didn't think I was being snarky; I thought I was being cautious!  ok, removed.